src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    };
+    attack() {
+        return this.strength
+    };
+    receiveDamage(damage) {
+        this.health = (this.health - damage);
+    };
+}
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength)
+        this.name = name;
+    };
+    receiveDamage(damage) {
+        this.health = (this.health - damage);
+        if (this.health <= 0) {
+            return `${this.name} has died in act of combat`
+        }
+        return `${this.name} has received ${damage} points of damage`
+    };
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength)
+    };
+    receiveDamage(damage) {
+        this.health = (this.health - damage);
+        if (this.health <= 0) {
+            return `A Saxon has died in combat`
+        }
+        return `A Saxon has received ${damage} points of damage`
+    };
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [],
+            this.saxonArmy = []
+    };
+    addViking(Soldier) {
+        this.vikingArmy.push(Soldier);
+    };
+    addSaxon(Soldier) {
+        this.saxonArmy.push(Soldier);
+    };
+    vikingAttack() {
+        let randomVik = this.vikingArmy[Math.floor(Math.random()) * this.vikingArmy.length];
+        let randomSax = this.saxonArmy[Math.floor(Math.random()) * this.saxonArmy.length];
+
+        let attack = (randomSax.receiveDamage(randomVik.attack()));
+
+        if (randomSax.health <= 0) {
+            this.saxonArmy.pop(randomSax)
```

Comment:
.pop() always removes the last element of the array, this is passing the test but you are not removing the Saxon that received the damage, for that you should have used the .splice()